packages/esm-reports-app/src/components/overview.component.tsx

```diff
@@ -0,0 +1,350 @@
+  TableHead,
+  TableHeader,
+  TableRow,
+} from '@carbon/react';
+import React, { useCallback, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { downloadMultipleReports, downloadReport, preserveReport, useReports } from './reports.resource';
+import {
+  ExtensionSlot,
+  isDesktop,
+  navigate,
+  showModal,
+  showToast,
+  useLayoutType,
+  userHasAccess,
+  useSession,
+} from '@openmrs/esm-framework';
+import { Calendar, Download, Play, Save, TrashCan } from '@carbon/react/icons';
+import styles from './reports.scss';
+import { DEFAULT_PAGE_NUMBER, DEFAULT_PAGE_SIZE, DEFAULT_PAGE_SIZES } from './pagination-constants';
+import { closeOverlay, launchOverlay } from '../hooks/useOverlay';
+import RunReportForm from './run-report/run-report-form.component';
+import Overlay from './overlay.component';
+import ReportStatus from './report-status.component';
+import { COMPLETED, FAILED, PROCESSING, RAN_REPORT_STATUSES, REQUESTED, SAVED } from './report-statuses-constants';
+import ReportOverviewButton from './report-overview-button.component';
+import { PRIVILEGE_SYSTEM_DEVELOPER } from '../constants';
+
+const OverviewComponent: React.FC = () => {
+  const { t } = useTranslation();
+  const currentSession = useSession();
+
+  let [checkedReportUuidsArray, setCheckedReportUuidsArray] = useState([]);
+  const [downloadReportButtonVisible, setDownloadReportButtonVisible] = useState(false);
+
+  useEffect(() => {
+    setDownloadReportButtonVisible(checkedReportUuidsArray.length > 0);
+  }, [checkedReportUuidsArray]);
+
+  const tableHeaders = [
+    { key: 'reportName', header: t('reportName', 'Report Name') },
+    { key: 'status', header: t('status', 'Status') },
+    { key: 'requestedBy', header: t('requestedBy', 'Requested by') },
+    { key: 'requestedOn', header: t('requestedOn', 'Requested on') },
+    { key: 'outputFormat', header: t('outputFormat', 'Output format') },
+    { key: 'parameters', header: t('parameters', 'Parameters') },
+    { key: 'actions', header: t('actions', 'Actions') },
+  ];
+
+  const [currentPage, setCurrentPage] = useState(DEFAULT_PAGE_NUMBER);
+  const [pageSize, setPageSize] = useState(DEFAULT_PAGE_SIZE);
+
+  const { reports, reportsTotalCount, mutateReports } = useReports(
+    RAN_REPORT_STATUSES.join(','),
+    currentPage,
+    pageSize,
+  );
+
+  const layout = useLayoutType();
+
+  function getReportStatus(row) {
+    return row?.cells.find((cell) => cell.info?.header === 'status')?.value;
+  }
+
+  function isCurrentUserTheSameAsReportRequestedByUser(reportRequestUuid: string) {
+    const report = reports.find((tableRow) => tableRow.id === reportRequestUuid);
+    const requestedByUserUuid = report?.requestedByUserUuid;
+    const currentUserUuid = currentSession?.user.uuid;
+
+    return requestedByUserUuid === currentUserUuid;
+  }
+
+  function isSystemDeveloperUser() {
+    return userHasAccess(PRIVILEGE_SYSTEM_DEVELOPER, currentSession.user);
+  }
```

Comment:
```suggestion
  function isSystemDeveloperUser(user: { roles: Array<Role> }) {
    return !isUndefined(user.roles.find((role) => role.display === PRIVILEGE_SYSTEM_DEVELOPER));
  }
```

Not sure if [this](https://github.com/harshthakkr/openmrs-esm-core/blob/main/packages/framework/esm-api/src/shared-api-objects/current-user.ts#L146-L148) works just as well.

Reply:
looks like it could work but I couldn't call this function as it not exposed (exported)